src/context/theme-provider.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+
+export interface ThemeProviderProps {
+  children: React.ReactNode
+}
+
+export type Theme = 'light' | 'dark'
+
+export type ThemeContextValues = {
+  currentTheme: Theme,
+  toggleTheme: () => void;
+  toggleThemeWithKey: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
+}
+
+const createThemeContext = () => React.createContext<ThemeContextValues | null>(null)
+export const ThemeContext = createThemeContext()
+
+export const ThemeProvider = ({ children }: ThemeProviderProps) => {
+  const [theme, setTheme] = React.useState<boolean>(() => {
+    const savedTheme =
+      typeof window !== 'undefined' && localStorage.getItem('theme')
+    if (savedTheme) return savedTheme === 'dark'
+    return window.matchMedia('prefers-color-scheme: dark').matches
+  })
+  React.useEffect(() => {
+    const htmlElem = document.documentElement
+    const currentTheme = theme ? 'dark' : 'light'
```

Comment:
seems like we can use better names for `theme` and `setTheme` since it seems to be a boolean indicating whether `dark` theme is selected or not

Reply:
perfect